src/graph_1.py

```diff
@@ -35,24 +39,21 @@ def del_node(self, val):
                     temp_edges[key] = self.edges[key]
             self.edges = temp_edges
         else:
-            raise IndexError("Node not in graph.")
+            raise Exception("Node not in graph.")
```

Comment:
This is just a non-descript exception. If they're searching for the node by name and that node doesn't exist, you should probably raise either a `ValueError` or a `KeyError`. This won't do